### DISCUSSION TITLE
Running on Rpi4

### DISCUSSION THREAD

**Discussion by Sarin-jacob** [zoriya/Kyoo#464] [Q&A] [answered]
I wanted to try this out, It looks awesome. Right now I'm running a Jellyfin instance on pi and I saw linux/arm64 listed in the containers. So I Have some questions.. 
- Can the transcoding be turned off / limited to certain very low impact transcoding.
- Does it create nfo files or writes to library directory (Where is metadata stored/ will it mess with existing jellyfin library)
- Does it support arm 32 bit
- Is there any recommended folder structure
- Minmum requirement for running an instance (hardware)
- I have watchtower setup, is there chance it would mess kyoo installation?

Thanks..

**Comment by zoriya** [comment, marked as answer]
Thanks for your interest in Kyoo! To answer your questions:

> Can the transcoding be turned off / limited to certain very low impact transcoding?

You can enable hardware transcoding, so it should be lighter on your machine. There is no way to completely turn off transcoding or limiting it. How would you see that in action? Would you want a toggle and simply see an error message if you can't transcode?


> Does it create nfo files or writes to library directory (Where is metadata stored/ will it mess with existing jellyfin library)?

The library is read only. It writes metadata on a database and download images on a docker volume. It won't mess your jellyfin library x)


> Does it support arm 32 bit?

There is no CD for this architecture but it should be trivial to add. Do you know the exact OS/ARCH tag for your machine?


> Is there any recommended folder structure?

No, the goal is to have kinda every folder structure work. There are still some items that fail to parse, if it does for you please open an issue or ask on the discord, this should get fixed!


> Minmum requirement for running an instance (hardware)?

No idea, I haven't tested this to be honest. Please report back!


> I have watchtower setup, is there chance it would mess kyoo installation?

As long as you use a tagged release and not master/edge/latest you'll be fine.

**Reply by Sarin-jacob** [reply]
> Would you want a toggle and simply see an error message if you can't transcode?

Yep, It seems Raspberry Pi 4  supports HW h264 decode n encoding and hevc decoding. bt I don't know how to enable that.. 
playing using pristine doesn't seem to affect the playback n its smooth too(1080p stream). changing it to auto or any other quality just make the playback stall and all the CPUs 100%. So its better to just disable those options.

> The library is read only. It writes metadata on a database and download images on a docker volume. It won't mess your jellyfin library x)

Cool.. I just saw the ro in the compose

> There are still some items that fail to parse, if it does for you please open an issue or ask on the discord, this should get fixed!

There are and some doesn't seem to find the metadata even when it gets the name right. it would be awesome if admins can add regex that can be used to remove part parsed name , is it possible?

> No idea, I haven't tested this to be honest. Please report back!

on Pristine settings it plays smoothly.  my jf library have about 17K video files, it would be great if there was some way to know if the matcher have scanned everything.  I will let u know tomorrow after everything is scanned and tested. :)

---
I had run into some minor hicups when I tried to install...
when I ran docker-compose up, at first I got 
`failed to register layer: archive/tar: invalid tar header` error
 then I tried docker-compose up a few more times till every image got pulled
for transcoder I got the error `layers from manifest don't match image configuration`
so I tried docker-compose up a few more times for some reason..
now its running.

**Reply by zoriya** [reply]
There is the list of parse/scan issue on the admin dashboard. What do you mean by "Remove part parsed name"? There is no plan on configuring regexes for parsing, everything should be automatic and if it's not, it should be fixed directly on kyoo's code instead of asking users to manually parse.

You have a section on how to enable hw transcode on the INSTALLING.md. You need to install drivers and change an env var and it should work. I am not sure on what's the expected behavior if you disable transcoding, but it's required to play the item (because your device can't play the codec, for example). Should we just display an error message?

**Reply by Sarin-jacob** [reply]
Many of the files comes with some sets of prefix, and for these some additional filter using regex would be helpful. 

 As for the transcoding issue.. Is it possible to add a toggle to hide the quality selection icon in the player ui. Because the default settings(pristine) works fine. 

Would just transmuxing aliiveate some of the problems with client compatibility? For me bandwidth is not an issue. 

When quality was set to anything other than original resolution n pristine, pi just hangs up. 

What does pristine actually do?

Thanks for answering my questions :)
Its time I get better hardware...

**Reply by zoriya** [reply]
Pristine is simply the original file being served with no processing it all. On the quality list, you have one marked as "Original". It's transmux, so the video stream is not changed, but it's moved from the original container (mkv, mp4...) to an hls container. If you select a quality other than the one marked original, it's transcode and the whole video stream is decoded/encoded in the specified quality.

Original is only visible on the list of qualities if its codec is supported by your client.

We could create a setting to disable transcoding, but that would mean being unable to play files with a video codec not supported by your client.

For the prefix, please join the discord or make an issue with 1/2 complete filepath as example. I'll fix the scanner to support them.

**Reply by Sarin-jacob** [reply]
A setting to disable transcoding would be nice, especially for older hardware that dosent support HW acelerated encoding.  For these, its either breaks the server or showing error message to the client. Being able to repurpose old hardware sure is nice.


I will open issues with proper logs and stuffs.

Thanks a lot.. You have done a great job on kyoo, I will tell my friends to use it since they think its a chore to maintain jellyfin. Once the filename detection is set properly it will be true setup and leave system.

**Reply by zoriya** [reply]
Thank you! I created an issue to track "disabling transcode": #468.
